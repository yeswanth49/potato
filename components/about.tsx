import { useEffect, useRef, useState } from 'react';

export function About() {
  const [isVisible, setIsVisible] = useState(false);
  const sectionRef = useRef<HTMLElement>(null);

  useEffect(() => {
    // In test environment, show animations immediately
    if (process.env.NODE_ENV === 'test') {
      setIsVisible(true);
      return;
    }

    const observer = new IntersectionObserver(
      ([entry]) => {
        if (entry.isIntersecting) {
          setIsVisible(true);
          observer.disconnect();
        }
      },
      { threshold: 0.1 }
    );

    if (sectionRef.current) {
      observer.observe(sectionRef.current);
    }

    return () => observer.disconnect();
  }, []);

  return (
    <section ref={sectionRef} id="about" aria-label="About section" className="px-4">
      <div className="max-w-3xl mx-auto">
        <div className={isVisible ? "motion-safe:animate-fade-in-up" : ""}>
          <h2 className="text-2xl md:text-3xl font-semibold mb-6 text-center">About</h2>
        </div>

        <div className={isVisible ? "motion-safe:animate-fade-in-up motion-safe:animate-delay-200" : ""}>
          <div className="prose max-w-none text-muted-foreground leading-relaxed">
            <p className="mb-4">
              I'm a frontend-focused developer with entrepreneurial drive, passionate about building accessible,
              performant web applications. My expertise spans React.js, Next.js, Node.js, and REST APIs, with a strong
              foundation in modern web technologies.
            </p>

            <p className="mb-4">
              With experience in cloud databases (Supabase, PostgreSQL, MongoDB), API testing, automated UI testing, and
              production deployment, I bring a comprehensive approach to web development. I have a solid understanding
              of data structures and algorithms, open-source collaboration, and the complete journey from idea to
              production.
            </p>

            <p>
              Currently pursuing my Bachelor of Technology in Computer Science and Engineering at Pragati Engineering
              College, I combine academic knowledge with practical experience in building real-world applications that
              serve thousands of users.
            </p>
          </div>
        </div>
      </div>
    </section>
  )
}
