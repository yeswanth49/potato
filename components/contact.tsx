import { Button } from "@/components/ui/button"
import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card"
import { Github, Linkedin, Mail, Phone } from "lucide-react"

export function Contact() {
  return (
    <section id="contact" aria-label="Contact section" className="px-4">
      <div className="max-w-3xl mx-auto">
        <div className="motion-safe:animate-fade-in-up">
          <h2 className="text-2xl md:text-3xl font-semibold mb-6 text-center">Get In Touch</h2>
        </div>

        <div className="motion-safe:animate-fade-in-up motion-safe:animate-delay-200">
          <Card className="border-border/50">
            <CardHeader className="text-center">
              <CardTitle className="text-xl mb-2">Let's Work Together</CardTitle>
              <CardDescription className="text-base leading-relaxed">
                I'm always interested in discussing new opportunities, innovative projects, and collaborations. Feel
                free to reach out if you'd like to connect!
              </CardDescription>
            </CardHeader>
            <CardContent>
              <div className="grid md:grid-cols-2 gap-4">
                <Button
                  variant="outline"
                  size="sm"
                  className="h-12 group hover:bg-transparent hover:text-foreground transition-colors bg-transparent"
                  asChild
                >
                  <a href="mailto:work.yeswanth@gmail.com" className="flex items-center gap-3">
                    <Mail className="w-4 h-4 group-hover:scale-110 transition-transform" />
                    <div className="text-left">
                      <div className="font-medium text-sm">Email</div>
                      <div className="text-xs text-muted-foreground group-hover:text-foreground/80">
                        work.yeswanth@gmail.com
                      </div>
                    </div>
                  </a>
                </Button>

                <Button
                  variant="outline"
                  size="sm"
                  className="h-12 group hover:bg-transparent hover:text-foreground transition-colors bg-transparent"
                  asChild
                >
                  <a href="tel:9032690321" className="flex items-center gap-3">
                    <Phone className="w-4 h-4 group-hover:scale-110 transition-transform" />
                    <div className="text-left">
                      <div className="font-medium text-sm">Phone</div>
                      <div className="text-xs text-muted-foreground group-hover:text-foreground/80">9032690321</div>
                    </div>
                  </a>
                </Button>

                <Button
                  variant="outline"
                  size="sm"
                  className="h-12 group hover:bg-transparent hover:text-foreground transition-colors bg-transparent"
                  asChild
                >
                  <a
                    href="https://github.com/yeswanth49"
                    target="_blank"
                    rel="noopener noreferrer"
                    className="flex items-center gap-3"
                  >
                    <Github className="w-4 h-4 group-hover:scale-110 transition-transform" />
                    <div className="text-left">
                      <div className="font-medium text-sm">GitHub</div>
                      <div className="text-xs text-muted-foreground group-hover:text-foreground/80">@yeswanth49</div>
                    </div>
                  </a>
                </Button>

                <Button
                  variant="outline"
                  size="sm"
                  className="h-12 group hover:bg-transparent hover:text-foreground transition-colors bg-transparent"
                  asChild
                >
                  <a
                    href="https://www.linkedin.com/in/yeswanth-madasu-41525b297/"
                    target="_blank"
                    rel="noopener noreferrer"
                    className="flex items-center gap-3"
                  >
                    <Linkedin className="w-4 h-4 group-hover:scale-110 transition-transform" />
                    <div className="text-left">
                      <div className="font-medium text-sm">LinkedIn</div>
                      <div className="text-xs text-muted-foreground group-hover:text-foreground/80">@yeswanthm</div>
                    </div>
                  </a>
                </Button>
              </div>
            </CardContent>
          </Card>
        </div>

        <div className="motion-safe:animate-fade-in-up motion-safe:animate-delay-400">
          <div className="text-center mt-8 pt-6 border-t border-border">
            <p className="text-muted-foreground text-sm">Â© {new Date().getFullYear()} Yeswanth Madasu. Built with Next.js and Tailwind CSS.</p>
          </div>
        </div>
      </div>
    </section>
  )
}
