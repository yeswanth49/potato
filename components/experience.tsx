import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card"
import { Badge } from "@/components/ui/badge"
import { ExternalLink } from "lucide-react"

const experiences = [
  {
    title: "Founder & Developer",
    company: "OpenBook",
    period: "2024 – Present",
    description:
      "Built AI-powered research/learning platform with Next.js (App Router) + Tailwind; streaming chat and rich Markdown/LaTeX",
    achievements: [
      "Orchestrated 5+ AI providers via AI SDK (tool-calling, structured outputs, server-side streaming)",
      "Implemented 3 study workflows (spaced repetition, memory palace, Feynman) and 2 web search tools (Exa, Tavily)",
      "Added rate limiting and structured logging middleware; instrumented 3 analytics layers (PostHog, Vercel Analytics, Speed Insights)",
      "Delivered PWA install experience; deployed on Vercel (serverless/edge) with streaming responses",
    ],
    technologies: ["Next.js", "React", "TypeScript", "Tailwind CSS", "AI SDK", "Vercel"],
    link: "https://goopenbook.in",
  },
  {
    title: "Founder & Developer",
    company: "safeLINK",
    period: "2024 – Present",
    description: "Mobile-first emergency QR profile app; generates QR-linked profiles for rapid access",
    achievements: [
      "SQLite + Jinja2 CRUD; server-side QR generation (qrcode/Pillow) with secure IDs; inline/downloadable PNGs",
      "In-browser QR scanner using WebRTC getUserMedia + jsQR; guided UX with modal prompts",
      "Exposed 8 routes (landing, template select, generate, profile, download, scanner, edit, verify)",
      "Password-gated edits with JSON verification; production-ready with Gunicorn",
    ],
    technologies: ["Flask", "SQLite", "Jinja2", "QR (qrcode/Pillow)", "WebRTC", "Gunicorn"],
  },
  {
    title: "Founder & Developer",
    company: "PEC.UP",
    period: "2023 – Present",
    description: "Resource-sharing platform with 1.5k+ registered users; handled 1–2k peak concurrent during exams",
    achievements: [
      "Implemented responsive layouts and performance optimizations for seamless cross-device UX",
      "Integrated REST APIs for dynamic content delivery and user management",
      "Successfully scaled to handle high concurrent user loads during peak exam periods",
    ],
    technologies: ["HTML", "CSS", "JavaScript", "Node.js"],
  },
]

export function Experience() {
  return (
    <section id="experience" className="px-4 bg-muted/30">
      <div className="max-w-5xl mx-auto">
        <div className="motion-safe:animate-fade-in-up">
          <h2 className="text-2xl md:text-3xl font-semibold mb-6 text-center">Professional Experience</h2>
        </div>

        <div className="space-y-6">
          {experiences.map((exp, index) => (
            <div
              key={`${exp.company}-${exp.title}`}
              className="motion-safe:animate-fade-in-up"
              style={{ animationDelay: `${(index + 1) * 120}ms` }}
            >
              <Card className="transition-colors border-border/50">
                <CardHeader>
                  <div className="flex flex-col md:flex-row md:items-center md:justify-between gap-3">
                    <div>
                      <CardTitle className="text-lg mb-1.5 flex items-center gap-2">
                        {exp.title} — {exp.company}
                        {exp.link ? (
                          <a
                            href={exp.link}
                            target="_blank"
                            rel="noopener noreferrer"
                            className="text-muted-foreground hover:text-foreground transition-colors"
                          >
                            <ExternalLink className="w-4 h-4" />
                          </a>
                        ) : null}
                      </CardTitle>
                      <CardDescription className="text-sm">{exp.period}</CardDescription>
                    </div>
                  </div>
                  <p className="text-muted-foreground leading-relaxed text-sm">{exp.description}</p>
                </CardHeader>
                <CardContent className="pt-2">
                  <ul className="space-y-1.5 mb-4">
                    {exp.achievements.map((achievement, i) => (
                      <li key={achievement} className="text-sm text-muted-foreground leading-relaxed flex items-start gap-2">
                        <span className="text-foreground mt-1" aria-hidden="true">•</span>
                        {achievement}
                      </li>
                    ))}
                  </ul>
                  <div className="flex flex-wrap gap-1.5 mt-1">
                    {exp.technologies.map((tech) => (
                      <Badge key={tech} variant="secondary" className="text-[10px] py-0.5">
                        {tech}
                      </Badge>
                    ))}
                  </div>
                </CardContent>
              </Card>
            </div>
          ))}
        </div>
      </div>
    </section>
  )
}
