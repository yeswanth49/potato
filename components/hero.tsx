'use client';

import { ArrowDown, Github, Linkedin, Twitter, FileText } from "lucide-react"
import { useCallback, useState, useEffect } from "react"

import { Button } from "@/components/ui/button"

export function Hero() {
  const [currentAge, setCurrentAge] = useState(() => {
    const birthDate = new Date('2006-01-02T00:00:00Z')
    const now = new Date()
    const diffTime = Math.abs(now.getTime() - birthDate.getTime())
    const diffYears = diffTime / (1000 * 60 * 60 * 24 * 365.25)
    return diffYears.toFixed(9)
  })

  useEffect(() => {
    const updateAge = () => {
      const birthDate = new Date('2006-01-02T00:00:00Z')
      const now = new Date()
      const diffTime = Math.abs(now.getTime() - birthDate.getTime())
      const diffYears = diffTime / (1000 * 60 * 60 * 24 * 365.25)
      setCurrentAge(diffYears.toFixed(9))
    }

    updateAge() // Initial calculation
    const interval = setInterval(updateAge, 100) // Update every 100ms for smooth counting

    return () => clearInterval(interval)
  }, [])

  const scrollToProjects = useCallback(() => {
    const projectsSection = document.getElementById('projects')
    if (projectsSection) {
      // Check if user prefers reduced motion
      const prefersReducedMotion = window.matchMedia('(prefers-reduced-motion: reduce)').matches

      if (prefersReducedMotion) {
        // Use instant scroll for users who prefer reduced motion
        projectsSection.scrollIntoView({
          behavior: 'auto',
          block: 'start',
          inline: 'nearest'
        })
        return
      }

      // Programmatic smooth scroll with custom duration and easing
      const startPosition = window.pageYOffset
      const targetPosition = projectsSection.offsetTop
      const distance = targetPosition - startPosition
      const duration = 4000 // 4 seconds
      let startTime: number | null = null

      function easeInOutCubic(t: number): number {
        return t < 0.5 ? 4 * t * t * t : (t - 1) * (2 * t - 2) * (2 * t - 2) + 1
      }

      function animateScroll(currentTime: number) {
        if (startTime === null) {
          startTime = currentTime
        }

        const timeElapsed = currentTime - startTime
        const progress = Math.min(timeElapsed / duration, 1)

        // Apply easing function
        const easeProgress = easeInOutCubic(progress)

        // Calculate new scroll position
        const newPosition = startPosition + (distance * easeProgress)

        window.scrollTo(0, newPosition)

        if (progress < 1) {
          requestAnimationFrame(animateScroll)
        }
      }

      requestAnimationFrame(animateScroll)
    }
  }, [])

  return (
    <section id="hero" className="flex items-center justify-center px-4">
      <div className="max-w-3xl mx-auto text-center">
        <div className="opacity-0 animate-fade-in-up">
        <h1 className="text-6xl md:text-7xl lg:text-8xl mb-3 font-league_spartan font-bold leading-none flex justify-center" style={{ fontFamily: 'var(--font-league-spartan)', fontWeight: '700', transform: 'translateX(2rem) translateY(4rem)' }}>
            <div className="flex flex-col">
              <span className="text-left">hi, i'm yeswanth.</span>
            </div>
          </h1>
        </div>


        <div className="opacity-0 animate-fade-in-up animate-delay-300">
          <p className="text-sm text-muted-foreground mb-8 max-w-2xl mx-auto leading-relaxed text-pretty" style={{ transform: 'translateX(2rem) translateY(4rem)' }}>
            been here for {currentAge} years
          </p>
        </div>
        <div className="opacity-0 animate-fade-in-up animate-delay-300">
          <p className="text-sm text-muted-foreground mb-8 max-w-2xl mx-auto leading-relaxed text-pretty" style={{ transform: 'translateX(2rem) translateY(3rem)' }}>
            trying to learn everything, by breaking everything.
          </p>
        </div>

        <div className="opacity-0 animate-fade-in-up animate-delay-400">
          <div className="flex flex-wrap items-center justify-center gap-6 mb-8" style={{ transform: 'translateX(2rem) translateY(4rem)' }}>
            <a
              href="https://x.com/Yeshh49"
              target="_blank"
              rel="noopener noreferrer"
              className="group flex items-center gap-2 text-muted-foreground hover:text-foreground transition-all duration-300 ease-in-out"
            >
              <Twitter className="w-5 h-5" />
              <span className="max-w-0 overflow-hidden group-hover:max-w-xs transition-all duration-300 ease-in-out whitespace-nowrap">
                Twitter
              </span>
            </a>
            <a
              href="https://www.linkedin.com/in/yeswanth49"
              target="_blank"
              rel="noopener noreferrer"
              className="group flex items-center gap-2 text-muted-foreground hover:text-foreground transition-all duration-300 ease-in-out"
            >
              <Linkedin className="w-5 h-5" />
              <span className="max-w-0 overflow-hidden group-hover:max-w-xs transition-all duration-300 ease-in-out whitespace-nowrap">
                LinkedIn
              </span>
            </a>
            <a
              href="https://github.com/yeswanth49"
              target="_blank"
              rel="noopener noreferrer"
              className="group flex items-center gap-2 text-muted-foreground hover:text-foreground transition-all duration-300 ease-in-out"
            >
              <Github className="w-5 h-5" />
              <span className="max-w-0 overflow-hidden group-hover:max-w-xs transition-all duration-300 ease-in-out whitespace-nowrap">
                GitHub
              </span>
            </a>
            <a
              href="/Yeswanth_Madasu.pdf"
              target="_blank"
              rel="noopener noreferrer"
              className="group flex items-center gap-2 text-muted-foreground hover:text-foreground transition-all duration-300 ease-in-out"
            >
              <FileText className="w-5 h-5" />
              <span className="max-w-0 overflow-hidden group-hover:max-w-xs transition-all duration-300 ease-in-out whitespace-nowrap">
                Resume
              </span>
            </a>
          </div>
        </div>

        <div className="opacity-0 animate-fade-in-up animate-delay-500">
          <Button
            variant="outline"
            size="sm"
            className="group transition-colors hover:bg-transparent hover:text-foreground bg-transparent"
            style={{ transform: 'translateX(2rem) translateY(4rem)' }}
            onClick={scrollToProjects}
          >
            Explore My Work
            <ArrowDown className="w-4 h-4 ml-2 group-hover:translate-y-0.5 transition-transform" />
          </Button>
        </div>
      </div>
    </section>
  )
}
