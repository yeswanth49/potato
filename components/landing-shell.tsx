"use client"

import { useState, useEffect, useRef, type ReactNode } from "react"
import dynamic from "next/dynamic"

import { KeyboardLanding } from "@/components/keyboard-landing"
import { cn } from "@/lib/utils"

const KeyboardBackgroundClient = dynamic(
  () => import("@/components/keyboard-background").then((mod) => ({ default: mod.KeyboardBackground })),
  { ssr: false },
)

interface LandingShellProps {
  children: ReactNode
  mainBackgroundMode?: "dark" | "keyboard"
  showLandingOnLoad?: boolean
}

export function LandingShell({
  children,
  mainBackgroundMode = "dark",
  showLandingOnLoad = true,
}: LandingShellProps) {
  const [showPortfolio, setShowPortfolio] = useState(!showLandingOnLoad)
  const [isTransitioning, setIsTransitioning] = useState(false)
  const [keyboardFadingOut, setKeyboardFadingOut] = useState(false)
  const [keyboardVisible, setKeyboardVisible] = useState(false)
  const transitionTimerRef = useRef<number | null>(null)

  useEffect(() => {
    if (showPortfolio) return

    const timer = window.setTimeout(() => {
      setKeyboardVisible(true)
    }, 100)

    return () => {
      window.clearTimeout(timer)
    }
  }, [showPortfolio])

  const handleCorrectEntry = () => {
    setIsTransitioning(true)
    setKeyboardFadingOut(true)

    // store timer id so we can clear it on unmount to avoid setState-after-unmount
    transitionTimerRef.current = window.setTimeout(() => {
      setShowPortfolio(true)
      // clear the ref after callback runs
      transitionTimerRef.current = null
    }, 500) as unknown as number
  }

  useEffect(() => {
    if (!showPortfolio) return

    const timer = window.setTimeout(() => {
      setIsTransitioning(false)
      setKeyboardFadingOut(false)
    }, 700)

    return () => {
      window.clearTimeout(timer)
    }
  }, [showPortfolio])

  // cleanup any outstanding transition timer on unmount
  useEffect(() => {
    return () => {
      if (transitionTimerRef.current) {
        window.clearTimeout(transitionTimerRef.current)
        transitionTimerRef.current = null
      }
    }
  }, [])

  if (!showPortfolio) {
    return (
      <div
        className={cn(
          keyboardFadingOut && "animate-fade-out",
          keyboardVisible && !keyboardFadingOut ? "animate-fade-in-keyboard" : "opacity-0",
        )}
      >
        <KeyboardLanding onCorrectEntry={handleCorrectEntry} backgroundMode={mainBackgroundMode} />
      </div>
    )
  }

  return (
    <main
      className={cn("min-h-screen bg-black relative", isTransitioning && "animate-fade-in-portfolio")}
      data-testid="portfolio-root"
    >
      {mainBackgroundMode === "keyboard" ? <KeyboardBackgroundClient /> : null}
      <div className="relative z-10">{children}</div>
    </main>
  )
}
