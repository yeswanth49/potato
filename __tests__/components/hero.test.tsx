/**
 * Testing framework: Jest + React Testing Library.
 * No existing test setup was found in package.json. These tests are authored for Jest + RTL.
 * If your project uses Vitest instead, replace jest.mock(...) with vi.mock(...),
 * provide equivalent setup to include '@testing-library/jest-dom', and configure path aliases.
 */
import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
// Helper to create icon mocks that forward all SVG props
const mockIcon = (testId: string) => {
  const IconComponent = (props: React.SVGProps<SVGSVGElement>) => (
    <svg data-testid={testId} {...props} />
  )
  IconComponent.displayName = `MockIcon(${testId})`
  return IconComponent
}

// Mock external modules virtually to avoid alias resolution issues in tests.
jest.mock('lucide-react', () => ({
  ArrowDown: mockIcon('arrow-down-icon'),
  Github: mockIcon('github-icon'),
  Linkedin: mockIcon('linkedin-icon'),
  Twitter: mockIcon('twitter-icon'),
  FileText: mockIcon('file-text-icon'),
}), { virtual: true })

import { Hero } from '@/components/hero'

jest.mock('@/components/ui/button', () => {
  const React = require('react')
  return {
    Button: ({ children, asChild, variant, size, className, ...props }: any) => {
      if (asChild && React.isValidElement(children)) {
        return React.cloneElement(children, {
          ...props,
          'data-testid': 'hero-button',
          'data-variant': variant,
          'data-size': size,
          className: className ? `${children.props.className || ''} ${className}`.trim() : children.props.className
        })
      }
      return React.createElement('button', {
        'data-testid': 'hero-button',
        'data-variant': variant,
        'data-size': size,
        className,
        ...props
      }, children)
    }
  }
}, { virtual: true })

describe('Hero Component', () => {
  beforeEach(() => {
    render(<Hero />)
  })

  describe('Structure', () => {
    it('renders the section with correct id and classes', () => {
      const section = document.getElementById('hero') as HTMLElement
      expect(section).toBeInTheDocument()
      expect(section.tagName).toBe('SECTION')
      expect(section).toHaveClass('flex', 'items-center', 'justify-center', 'px-4')
    })

    it('contains a centered max width container', () => {
      const section = document.getElementById('hero') as HTMLElement
      const container = section.querySelector('.max-w-3xl') as HTMLElement
      expect(container).toBeInTheDocument()
      expect(container).toHaveClass('mx-auto', 'text-center')
    })
  })

  describe('Primary content', () => {
    it('renders main heading text', () => {
      const h1 = screen.getByRole('heading', { level: 1 })
      expect(h1).toHaveTextContent("Hi, I'm Yeswanth Madasu")
      expect(h1).toHaveClass('text-4xl', 'md:text-5xl', 'font-semibold', 'mb-3', 'text-balance')
    })

    it('renders subtitle', () => {
      const subtitle = screen.getByText('Full Stack Developer')
      expect(subtitle).toBeInTheDocument()
      expect(subtitle.tagName).toBe('P')
      expect(subtitle).toHaveClass('text-base', 'md:text-lg', 'text-muted-foreground', 'mb-4', 'text-balance')
    })

    it('renders description with both sentences', () => {
      const description = screen.getByText(/Frontend-focused developer/)
      expect(description).toBeInTheDocument()
      expect(description).toHaveTextContent(/Frontend-focused developer building accessible/)
      expect(description).toHaveTextContent(/Focusing on building accessible/)
      expect(description).toHaveClass(
        'text-sm',
        'text-muted-foreground',
        'mb-8',
        'max-w-2xl',
        'mx-auto',
        'leading-relaxed',
        'text-pretty'
      )
    })
  })

  describe('Animation wrappers', () => {
    it('heading wrapper has fade-in classes', () => {
      const h1 = screen.getByRole('heading', { level: 1 })
      const wrapper = h1.parentElement as HTMLElement
      expect(wrapper).toHaveClass('opacity-0', 'animate-fade-in-up')
    })

    it('staggered delay on subtitle and description', () => {
      const subtitleWrapper = screen.getByText('Full Stack Developer').parentElement as HTMLElement
      expect(subtitleWrapper).toHaveClass('opacity-0', 'animate-fade-in-up', 'animate-delay-200')

      const descriptionWrapper = screen.getByText(/Frontend-focused developer/).parentElement as HTMLElement
      expect(descriptionWrapper).toHaveClass('opacity-0', 'animate-fade-in-up', 'animate-delay-300')
    })

    it('social container and CTA have delayed animations', () => {
      // Find the main social links container by looking for the flex container that contains all social links
      const socialLinks = screen.getAllByRole('link')
      const socialContainer = socialLinks[0].parentElement?.parentElement as HTMLElement
      expect(socialContainer).toHaveClass('opacity-0', 'animate-fade-in-up', 'animate-delay-400')

      const button = screen.getByTestId('hero-button')
      const buttonWrapper = button.parentElement as HTMLElement
      expect(buttonWrapper).toHaveClass('opacity-0', 'animate-fade-in-up', 'animate-delay-500')
    })
  })

  describe('Social links', () => {
    it('has external profile links with secure attributes', () => {
      const twitter = screen.getByRole('link', { name: /twitter/i })
      expect(twitter).toHaveAttribute('href', 'https://x.com/Yeshh49')
      expect(twitter).toHaveAttribute('target', '_blank')
      expect(twitter).toHaveAttribute('rel', 'noopener noreferrer')

      const github = screen.getByRole('link', { name: /github/i })
      expect(github).toHaveAttribute('href', 'https://github.com/yeswanth49')
      expect(github).toHaveAttribute('target', '_blank')
      expect(github).toHaveAttribute('rel', 'noopener noreferrer')

      const linkedin = screen.getByRole('link', { name: /linkedin/i })
      expect(linkedin).toHaveAttribute('href', 'www.linkedin.com/in/yeswanth49')
      expect(linkedin).toHaveAttribute('target', '_blank')
      expect(linkedin).toHaveAttribute('rel', 'noopener noreferrer')
    })


    it('resume link opens in new tab securely', () => {
      const resume = screen.getByRole('link', { name: /resume/i })
      expect(resume).toHaveAttribute('href', '/Yeswanth_Madasu.pdf')
      expect(resume).toHaveAttribute('target', '_blank')
      expect(resume).toHaveAttribute('rel', 'noopener noreferrer')
    })

    it('social links have expected classes', () => {
      const twitter = screen.getByRole('link', { name: /twitter/i })
      expect(twitter).toHaveClass(
        'group',
        'flex',
        'items-center',
        'gap-2',
        'text-muted-foreground',
        'hover:text-foreground',
        'transition-all',
        'duration-300',
        'ease-in-out'
      )
      const label = screen.getByText('Twitter')
      expect(label).toHaveClass(
        'max-w-0',
        'overflow-hidden',
        'group-hover:max-w-xs',
        'transition-all',
        'duration-300',
        'ease-in-out',
        'whitespace-nowrap'
      )
    })
  })

  describe('Icons', () => {
    it('renders all icons', () => {
      expect(screen.getByTestId('twitter-icon')).toBeInTheDocument()
      expect(screen.getByTestId('github-icon')).toBeInTheDocument()
      expect(screen.getByTestId('linkedin-icon')).toBeInTheDocument()
      expect(screen.getByTestId('file-text-icon')).toBeInTheDocument()
    })

    it('applies correct dimensions on representative icons', () => {
      expect(screen.getByTestId('twitter-icon')).toHaveClass('w-5', 'h-5')
      expect(screen.getByTestId('github-icon')).toHaveClass('w-5', 'h-5')
    })
  })

  describe('CTA Button', () => {
    it('renders CTA with text and icon', () => {
      const button = screen.getByTestId('hero-button')
      expect(button).toBeInTheDocument()
      expect(button).toHaveTextContent(/Explore My Work/)
      expect(screen.getByTestId('arrow-down-icon')).toBeInTheDocument()
    })

    it('applies variant/size and styling classes', () => {
      const button = screen.getByTestId('hero-button')
      expect(button).toHaveAttribute('data-variant', 'outline')
      expect(button).toHaveAttribute('data-size', 'sm')
      expect(button).toHaveClass('group', 'transition-colors', 'hover:bg-transparent', 'hover:text-foreground', 'bg-transparent')
    })

    it('contains anchor with correct href', () => {
      const link = screen.getByRole('link', { name: /explore my work/i })
      expect(link).toHaveAttribute('href', '#about')
    })

    it('icon has hover translate and transition classes', () => {
      const icon = screen.getByTestId('arrow-down-icon')
      expect(icon).toHaveClass('w-4', 'h-4', 'ml-2', 'group-hover:translate-y-0.5', 'transition-transform')
    })
  })

  describe('Responsive and animation class patterns', () => {
    it('heading and subtitle have responsive text classes', () => {
      expect(screen.getByRole('heading', { level: 1 })).toHaveClass('text-4xl', 'md:text-5xl')
      expect(screen.getByText('Full Stack Developer')).toHaveClass('text-base', 'md:text-lg')
    })

    it('social links container has layout classes', () => {
      // Find the main social links container by looking for the flex container that contains all social links
      const socialLinks = screen.getAllByRole('link')
      const container = socialLinks[0].parentElement as HTMLElement
      expect(container).toHaveClass('flex', 'flex-wrap', 'items-center', 'justify-center', 'gap-6', 'mb-8')
    })

    it('every animated wrapper includes at least one animate-* class', () => {
      const animatedEls = (document.getElementById('hero') as HTMLElement).querySelectorAll('[class*="animate-"]')
      animatedEls.forEach(el => {
        const hasAnimate = Array.from(el.classList).some(c => c.startsWith('animate-'))
        expect(hasAnimate).toBe(true)
      })
    })
  })

  describe('Content sanity checks', () => {
    it('avoids placeholder text and maintains professional tone', () => {
      const text = (document.getElementById('hero') as HTMLElement).textContent || ''
      expect(text).not.toMatch(/lorem ipsum/i)
      expect(text).not.toMatch(/placeholder/i)
      expect(text).toMatch(/React and Next\.js/)
    })

  })
})

describe('Hero Snapshot', () => {
  it('matches DOM snapshot', () => {
    const { container } = render(<Hero />)
    expect(container.firstChild).toMatchSnapshot()
  })
})