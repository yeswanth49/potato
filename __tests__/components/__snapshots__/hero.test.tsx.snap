// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Hero Snapshot matches DOM snapshot 1`] = `
<section
  class="flex items-center justify-center px-4"
  id="hero"
>
  <div
    class="max-w-3xl mx-auto text-center"
  >
    <div
      class="opacity-0 animate-fade-in-up"
    >
      <h1
        class="text-4xl md:text-5xl font-semibold mb-3 text-balance"
      >
        Hi, I'm Yeswanth Madasu
      </h1>
    </div>
    <div
      class="opacity-0 animate-fade-in-up animate-delay-200"
    >
      <p
        class="text-base md:text-lg text-muted-foreground mb-4 text-balance"
      >
        Full Stack Developer
      </p>
    </div>
    <div
      class="opacity-0 animate-fade-in-up animate-delay-300"
    >
      <p
        class="text-sm text-muted-foreground mb-8 max-w-2xl mx-auto leading-relaxed text-pretty"
      >
        Frontend-focused developer building accessible, performant web apps with React and Next.js. Focusing on building accessible, performant web apps with React and Next.js.
      </p>
    </div>
    <div
      class="opacity-0 animate-fade-in-up animate-delay-400"
    >
      <div
        class="flex flex-wrap items-center justify-center gap-6 mb-8"
      >
        <a
          class="group flex items-center gap-2 text-muted-foreground hover:text-foreground transition-all duration-300 ease-in-out"
          href="https://x.com/Yeshh49"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="w-5 h-5"
            data-testid="twitter-icon"
          />
          <span
            class="max-w-0 overflow-hidden group-hover:max-w-xs transition-all duration-300 ease-in-out whitespace-nowrap"
          >
            Twitter
          </span>
        </a>
        <a
          class="group flex items-center gap-2 text-muted-foreground hover:text-foreground transition-all duration-300 ease-in-out"
          href="https://www.linkedin.com/in/yeswanth-madasu-41525b297/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="w-5 h-5"
            data-testid="linkedin-icon"
          />
          <span
            class="max-w-0 overflow-hidden group-hover:max-w-xs transition-all duration-300 ease-in-out whitespace-nowrap"
          >
            LinkedIn
          </span>
        </a>
        <a
          class="group flex items-center gap-2 text-muted-foreground hover:text-foreground transition-all duration-300 ease-in-out"
          href="https://github.com/yeswanth49"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="w-5 h-5"
            data-testid="github-icon"
          />
          <span
            class="max-w-0 overflow-hidden group-hover:max-w-xs transition-all duration-300 ease-in-out whitespace-nowrap"
          >
            GitHub
          </span>
        </a>
        <a
          class="group flex items-center gap-2 text-muted-foreground hover:text-foreground transition-all duration-300 ease-in-out"
          href="/Yeswanth_Madasu.pdf"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="w-5 h-5"
            data-testid="file-text-icon"
          />
          <span
            class="max-w-0 overflow-hidden group-hover:max-w-xs transition-all duration-300 ease-in-out whitespace-nowrap"
          >
            Resume
          </span>
        </a>
      </div>
    </div>
    <div
      class="opacity-0 animate-fade-in-up animate-delay-500"
    >
      <a
        class="group transition-colors hover:bg-transparent hover:text-foreground bg-transparent"
        data-size="sm"
        data-testid="hero-button"
        data-variant="outline"
        href="#about"
      >
        Explore My Work
        <svg
          class="w-4 h-4 ml-2 group-hover:translate-y-0.5 transition-transform"
          data-testid="arrow-down-icon"
        />
      </a>
    </div>
  </div>
</section>
`;
